listenbrainz_spark/tests/test_utils.py

```diff
@@ -9,79 +9,71 @@
 from pyspark.sql import Row
 
 class UtilsTestCase(SparkTestCase):
+    # use path_ as prefix for all paths in this class.
+    path_ = '/test'
 
-    def test_append_dataframe(self):
-        path_ = 'test_df.parquet'
-        hdfs_path = os.path.join(config.HDFS_CLUSTER_URI, path_)
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        utils.delete_dir(self.path_, recursive=True)
 
-        df = utils.create_dataframe(Row(column1=1, column2=2), schema=None)
+    def test_append_dataframe(self):
+        hdfs_path = self.path_ + '/test_df.parquet'
+        df = utils.create_dataframe([Row(column1=1, column2=2)], schema=None)
         utils.append(df, hdfs_path)
         new_df = utils.read_files_from_HDFS(hdfs_path)
         self.assertEqual(new_df.count(), 1)
 
-        df = utils.create_dataframe(Row(column1=3, column2=4), schema=None)
+        df = utils.create_dataframe([Row(column1=3, column2=4)], schema=None)
         utils.append(df, hdfs_path)
         appended_df = utils.read_files_from_HDFS(hdfs_path)
         self.assertEqual(appended_df.count(), 2)
 
     def test_create_dataframe(self):
-        path_ = 'test_df.parquet'
-        hdfs_path = os.path.join(config.HDFS_CLUSTER_URI, path_)
-
-        df = utils.create_dataframe(Row(column1=1, column2=2), schema=None)
+        hdfs_path = self.path_ + '/test_df.parquet'
+        df = utils.create_dataframe([Row(column1=1, column2=2)], schema=None)
         self.assertEqual(df.count(), 1)
         utils.save_parquet(df, hdfs_path)
 
         received_df = utils.read_files_from_HDFS(hdfs_path)
         self.assertEqual(received_df.count(), 1)
 
     def test_create_dir(self):
-        path_ = '/tests/test'
-        utils.create_dir(path_)
-        status = utils.path_exists(path_)
+        utils.create_dir(self.path_)
+        status = utils.path_exists(self.path_)
         self.assertTrue(status)
 
     def test_delete_dir(self):
-        path_ = '/tests/test'
-        utils.create_dir(path_)
-
-        utils.delete_dir(path_)
-        status = utils.path_exists(path_)
+        utils.create_dir(self.path_)
+        utils.delete_dir(self.path_)
+        status = utils.path_exists(self.path_)
         self.assertFalse(status)
+        # for tearDown method.
+        utils.create_dir(self.path_)
```

Comment:
This does not sound like a good practice, other tests will also need to create dir even if they don't need it. Let's fix the teardown function to only delete if the dir exists and do ko thing otherwise.